test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusDevModeTest.java

```diff
@@ -404,10 +404,27 @@ public void addResourceFile(String path, byte[] data) {
      */
     public void deleteResourceFile(String path) {
         final Path resourceFilePath = deploymentResourcePath.resolve(path);
-        try {
-            Files.delete(resourceFilePath);
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+        long timeout = System.currentTimeMillis() + 5000;
+        //in general there is a potential race here
+        //if you serve a file you will send the data to the client, then close the resource
+        //this means that by the time the client request is run the file may not 
+        //have been closed yet, as the test sees the response as being complete after the last data is send
+        //we wait up to 5s for this condition to be resolved
+        for (;;) {
+            try {
+                Files.delete(resourceFilePath);
+                break;
+            } catch (IOException e) {
+                if (System.currentTimeMillis() < timeout) {
+                    continue;
+                }
+                try {
+                    Thread.sleep(50);
+                } catch (InterruptedException ex) {
+                    //ignore
+                }
```

Comment:
I don't really understand this, why sleep when the timeout has been exceeded and not just throw the exception right away?

Reply:
Oops, I have them in the wrong order

Reply:
OK :). I'll approve as soon as it's fixed